src/segment_evaluation.rs

```diff
@@ -19,125 +19,156 @@ use crate::{
     entity::{AttrValue, Entity},
     models::TargetingRule,
 };
+use crate::errors::{Result, Error};
+
+// For chaining errors creating useful error messages:
+use anyhow::{Context, Result as AnyhowResult};
+use anyhow::anyhow;
```

Comment:
We don't really want `anyhow` in library code. It hides most of information about the actual error. Let's reserve `anyhow` for application.